yt_dlp/extractor/qqmusic.py

```diff
@@ -1,245 +1,314 @@
+        },
+        'playlist_count': 12,
+        'playlist': [{
+            'info_dict': {
+                'id': '0016cvsy02mmCl',
+                'ext': 'mp3',
+                'title': '群青',
+                'album': '桃几2021年翻唱集',
+                'release_date': '20210913',
+                'duration': 248,
+                'creators': ['桃几OvO'],
+                'genres': ['Pop'],
+                'description': 'md5:4296005a04edcb5cdbe0889d5055a7ae',
+                'size': 3970822,
+                'thumbnail': r're:^https?://.*\.jpg(?:$|[#?])',
+            },
+        }],
+    }]
+
+    def _entries(self, mid, init_data):
+        page_size = 50
+        max_num = traverse_obj(init_data, ('singerDetail', 'songTotalNum'))
+        for page in range(max_num // page_size + 1):
+            data = self.make_fcu_req({'req_1': {
+                'module': 'music.web_singer_info_svr',
+                'method': 'get_singer_detail_info',
+                'param': {
+                    'sort': 5,
+                    'singermid': mid,
+                    'sin': page * page_size,
+                    'num': page_size,
+                }}}, mid, note=f'Downloading page {page}')
+            yield from traverse_obj(data, ('req_1', 'data', 'songlist', ..., {lambda x: self.url_result(
+                f'https://y.qq.com/n/ryqq/songDetail/{x["mid"]}', QQMusicIE, x['mid'], x.get('title'))}))
 
     def _real_extract(self, url):
         mid = self._match_id(url)
 
-        entries = self.get_entries_from_page(mid)
-        singer_page = self._download_webpage(url, mid, 'Download singer page')
-        singer_name = self._html_search_regex(
-            r"singername\s*:\s*'(.*?)'", singer_page, 'singer name', default=None)
-        singer_desc = None
+        init_data = self.download_init_data(url, mid)
+        return self.playlist_result(
+            self._entries(mid, init_data), mid, **traverse_obj(init_data, ('singerDetail', {
+                'title': ('basic_info', 'name', {str}),
+                'description': ('ex_info', 'desc', {str}),
+                'thumbnail': ('pic', 'pic', {url_or_none}),
+            })))
 
-        if mid:
-            singer_desc_page = self._download_xml(
-                'http://s.plcloud.music.qq.com/fcgi-bin/fcg_get_singer_desc.fcg', mid,
-                'Donwload singer description XML',
-                query={'utf8': 1, 'outCharset': 'utf-8', 'format': 'xml', 'singermid': mid},
-                headers={'Referer': 'https://y.qq.com/n/yqq/singer/'})
 
-            singer_desc = singer_desc_page.find('./data/info/desc').text
-
-        return self.playlist_result(entries, mid, singer_name, singer_desc)
+class QQPlaylistBaseIE(InfoExtractor):
+    def _extract_entries(self, info_json, path):
+        return traverse_obj(info_json, (*path, {lambda song: self.url_result(
+            f'https://y.qq.com/n/ryqq/songDetail/{song["songmid"]}', QQMusicIE, song['songmid'], song['songname'])}))
```

Comment:
I would use a generator here
```suggestion
    def _extract_entries(self, info_json, path):
        for entry in traverse_obj(info_json, *path):
            song_mid = song['songmid']
            yield self.url_result(
                f'https://y.qq.com/n/ryqq/songDetail/{song_mid}',
                QQMusicIE, song_mid, song.get('songname'))
```